virtualstack.go

```diff
@@ -0,0 +1,16 @@
+package failure
+
+type VirtualStack interface {
+	Push(v interface{})
+}
+
+func AsVirtualStack(err error, vs VirtualStack) {
```

Comment:
This name should be `AsEach` or something for iterating every error by calling `As` method.

Reply:
Abstraction can be done when it is necessary afterwards.